azure-pipelines/release.yml

```diff
@@ -17,64 +17,70 @@ variables:
 - group: vscode-csharp release secrets
 
 jobs:
-- job: PublishToMarketplace
+- deployment: PublishToMarketplace
```

Comment:
Using a deployment job with an environment that requires approval seems to be the only way to enable manual approvals in a yaml pipeline.  See https://stackoverflow.com/questions/61656077/implementing-gates-in-azure-yaml-pipelines